src/odemis/util/spectrum.py

```diff
@@ -55,7 +55,11 @@ def get_wavelength_per_pixel(da):
         wl = da.metadata[model.MD_WL_LIST]
         # check dimension of data
         dims = da.metadata.get(model.MD_DIMS, "CTZYX"[-da.ndim:])
-        if len(dims) == 3 and dims == "YXC" and da.shape[2] in (3, 4):  # RGB?
+        if len(da.shape) == 2:
+            # This is a hack to make the line spectrum select the right dimension and report
```

Comment:
This is for raw line spectrum? In this case, the MD_DIMS should be "XC", and the rest should work normally.
=> Fix the .projectAsRaw() in the stream, and no need to update the function here.